Snakefile

```diff
@@ -19,193 +74,207 @@ rule index_sequences:
         sequences = input_fasta
     output:
         sequence_index = "results/sequence_index.tsv"
+    params:
+        verbose = '--verbose' if config['verbose'] else '',
     shell:
         """
         augur index \
             --sequences {input.sequences} \
             --output {output.sequence_index}
+            {params.verbose}
         """
 
 rule filter:
     message:
         """
-        Filtering to
-          - {params.sequences_per_group} sequence(s) per {params.group_by!s}
-          - from {params.min_date} onwards
-          - excluding strains in {input.exclude}
+        Filtering strains
         """
     input:
+        excluded_strains = excluded_strains,
+        included_strains = included_strains,        
+        metadata = input_metadata,
         sequences = input_fasta,
         sequence_index = "results/sequence_index.tsv",
-        metadata = input_metadata,
-        exclude = dropped_strains
     output:
+        log = "logs/filter.log",
         sequences = "results/filtered.fasta"
     params:
-        group_by = "country year month",
-        sequences_per_group = 20,
-        min_date = 2012
+        allParams = format_config_params('filter')
```

Comment:
I understand the need for a function to convert the config to command-line options, but I do think we lose some legibility by the additional indirection here of doing them all together instead of individually.  If we decided it was ok to not support arbitrary options, then we could list out the params for a rule to be explicit about their mapping to config (and vice versa), e.g.:

```python
params:
    min_date = augur_option(config["filter"]["--min-date"]),
    max_date = augur_option(config["filter"]["--max-date"]),
    …
```

or even:

```python
params:
    **augur_options(
        config["filter"],
        "--min-date",
        "--max-date",
        ...
    )
```
This also greatly increases greppability of config option usage.